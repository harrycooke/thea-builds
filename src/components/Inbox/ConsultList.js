import React from 'react';
import PropTypes from 'prop-types';
import { connect } from "react-redux";
import { makeStyles } from '@material-ui/core/styles';
import _ from "../../@lodash";
import {
  Table,
  TableBody,
  TableCell,
  TablePagination,
  TableRow,
  TableHead,
  Paper,
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogContentText,
  DialogTitle,
  Slide  
} from "@material-ui/core";
import { fetchUserData } from "../../actions/authActions";
import { modifyCase } from "../../actions/caseAction";

const Transition = React.forwardRef(function Transition(props, ref) {
  return <Slide direction="up" ref={ref} {...props} />;
});

function desc(a, b, orderBy) {
  if (b[orderBy] < a[orderBy]) {
    return -1;
  }
  if (b[orderBy] > a[orderBy]) {
    return 1;
  }
  return 0;
}

function stableSort(array, cmp) {
  const stabilizedThis = array.map((el, index) => [el, index]);
  stabilizedThis.sort((a, b) => {
    const order = cmp(a[0], b[0]);
    if (order !== 0) return order;
    return a[1] - b[1];
  });
  return stabilizedThis.map(el => el[0]);
}

function getSorting(order, orderBy) {
  return order === 'desc' ? (a, b) => desc(a, b, orderBy) : (a, b) => -desc(a, b, orderBy);
}

const useStyles = makeStyles(theme => ({
  root: {
    width: '100%',
    marginTop: theme.spacing(3),
  },
  paper: {
    width: '100%',
    marginBottom: theme.spacing(2),
    boxShadow: "none",
  },
  table: {
    // minWidth: 750,
  },
  tableWrapper: {
    overflowX: 'auto',
  },
  tableCellBold: {
    color: "#1C1C1C",
    fontSize: 20,
    fontWeight: 500,
  },
  tableCellLight: {
    color: "#1c1c1c",
    fontSize: 18,
    fontWeight: "normal"
  }
}));

//This is not that imp in terms of testing

export function ConsultList(props) {
  const classes = useStyles();
  const [order] = React.useState('desc');
  const [orderBy] = React.useState('id');
  const [page, setPage] = React.useState(0);
  const [dense] = React.useState(false);
  const [rowsPerPage, setRowsPerPage] = React.useState(10);
  const [selectedConsult, setSelected] = React.useState(props.selectedConsult);
  const [selectedtemp, setSelectedtemp] = React.useState(null);
  const [emailTo, setEmailTo] = React.useState(null);

  let rows = [], rowsLength = 0;
  if (props.rows) {
    rows = props.rows;
    rowsLength = rows.length;
  }
  //Confirm Dialog 
  const [open, setOpen] = React.useState(false);
  const [unassignmodal, setUnassignmodal] = React.useState(false);

  const handleConfirmDialog = (e, id, emailOfSpecialist) => {
    setSelectedtemp(id);
    setEmailTo(emailOfSpecialist);
    setOpen(true);
  };

  const handleClose = () => {
    setEmailTo(null);
    setOpen(false);
  };

  const assignConsultant = (e, consultId, emailTo, caseData) => {
    if(consultId){
      const { case_id } = props.params;
      const current_time = new Date();
      const current_timestamp = current_time.getTime();
      const newCaseData = {
        caseId: case_id,
        last_updated_by_id: props.userdata.id,
        updated_at: current_timestamp,
        specialist_id: consultId,
        emailTo: emailTo
      };
      let readByOthers = caseData.is_read_by_other ? caseData.is_read_by_other : [];
      let ifIndex = _.indexOf(readByOthers, props.userdata.id);
      if(ifIndex < 0){
        readByOthers.push(props.userdata.id);
        newCaseData['is_read_by_other'] = readByOthers;
      }
      props.modifyCase(newCaseData);
      setSelected(consultId);
      setEmailTo(null);
      setOpen(false);
    }
  };

  const handleUnassignDialog = (e, id) => {
    setSelectedtemp(id);
    setUnassignmodal(true);
    setOpen(true);
  };

  const unassignConsultant = (e, consultId ) => {
    if(consultId === 0){
      const { case_id } = props.params;
      const current_time = new Date();
      const current_timestamp = current_time.getTime();
      const newCaseData = {
        caseId: case_id,
        last_updated_by_id: props.userdata.id,
        updated_at: current_timestamp,
        specialist_id: consultId
      };
      props.modifyCase(newCaseData);
      setSelected(null);
      setUnassignmodal(false);
      setOpen(false);
    }
  }  

  const handleUnassignClose = () => {
    setUnassignmodal(false);
    setOpen(false);
  };

  function handleChangePage(event, newPage) {
    setPage(newPage);
  }

  function handleChangeRowsPerPage(event) {
    setRowsPerPage(+event.target.value);
  }

  return (
    <div className={classes.root}>
      <Paper className={classes.paper}>
        <div className={classes.tableWrapper}>
          <Table
            className={classes.table}
            aria-labelledby="tableTitle"
            size={dense ? 'small' : 'medium'}
          >
            <TableHead>
              <TableRow>
                <TableCell padding="checkbox" align="center">Select Specialist</TableCell>
                <TableCell align="center">First Name</TableCell>
                <TableCell align="center">Last Name</TableCell>
                <TableCell align="center">Specialty</TableCell>
              </TableRow>
            </TableHead>
            <TableBody>
              {stableSort(rows, getSorting(order, orderBy))
                .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
                .map((row, index) => {
                  return (

                    <TableRow
                      hover
                      selected={selectedConsult === row.id ? true: false}
                      tabIndex={-1}
                      key={index}
                    >
                      <TableCell align="center" padding="checkbox">
                        <div>
                        {selectedConsult === row.id ?
                          <Button variant="contained" color="primary" onClick={event => handleUnassignDialog(event, 0)}>Unassign</Button>
                          :
                          <Button variant="outlined" onClick={event => handleConfirmDialog(event, row.id, row.email)}>Assign</Button>
                        }
                        </div>
                      </TableCell>

                      <TableCell align="center" className={classes.tableCellBold}>
                        {row.firstName ? row.firstName : null}
                      </TableCell>

                      
                      <TableCell
                        align="center"
                        className={classes.tableCellBold}
                      >
                        {row.lastName ? row.lastName : null}
                      </TableCell>
                      
                      <TableCell align="center"  className={classes.tableCellBold}>
                      {row.specialty ? row.specialty : null}
                      </TableCell>
                    </TableRow>
                  );
                })}

              {rowsLength === 0 && (
                <TableRow style={{ height: 49 }}>
                  <TableCell colSpan={6}> No Consult.. </TableCell>
                </TableRow>
              )}
            </TableBody>
          </Table>
          <Dialog
            open={open}
            TransitionComponent={Transition}
            keepMounted
            onClose={unassignmodal ? handleUnassignClose: handleClose }
            aria-labelledby="alert-dialog-slide-title"
            aria-describedby="alert-dialog-slide-description"
          >
            <DialogTitle id="alert-dialog-slide-title">{"Please confirm"}</DialogTitle>
            <DialogContent>
              <DialogContentText id="alert-dialog-slide-description">
                Are you sure you want to {unassignmodal ?  'unassign': 'assign'} this specialist {unassignmodal ?  'from': 'to'} {props.patientName}?
              </DialogContentText>
            </DialogContent>
            {unassignmodal ?
              <DialogActions>
                <Button onClick={handleUnassignClose} color="primary">
                  No
                </Button>
                <Button onClick={event => unassignConsultant(event, selectedtemp)} color="primary">
                  Yes
                </Button>
              </DialogActions>
            :
              <DialogActions>
                <Button onClick={handleClose} color="primary">
                  No
                </Button>
                <Button onClick={event => assignConsultant(event, selectedtemp, emailTo, props.currentCaseData)} color="primary">
                  Yes
                </Button>
              </DialogActions>
            }
          </Dialog>
        </div>
        <TablePagination
          rowsPerPageOptions={[5, 10, 15]}
          component="div"
          count={rows.length}
          rowsPerPage={rowsPerPage}
          page={page}
          backIconButtonProps={{
            'aria-label': 'Previous Page',
          }}
          nextIconButtonProps={{
            'aria-label': 'Next Page',
          }}
          onChangePage={handleChangePage}
          onChangeRowsPerPage={handleChangeRowsPerPage}
        />
      </Paper>
    </div>
  );
}

ConsultList.propsTypes = {
  fetchUserData: PropTypes.func.isRequired,
  userdata: PropTypes.object.isRequired,
  modifyCase: PropTypes.func.isRequired
}

const mapStateToProps = state => ({
  userdata: state.auth.userdata
})

export default connect(
  mapStateToProps,
  { fetchUserData, modifyCase }
)(ConsultList)
