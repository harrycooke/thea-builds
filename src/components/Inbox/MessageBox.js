import React, { Component } from 'react';
import { withStyles } from '@material-ui/core/styles';

const styles = theme => ({

});

export class MessageBox extends Component {
  render() {
    return (
      <div></div>
    );
  }
}

export default withStyles(styles)(MessageBox);