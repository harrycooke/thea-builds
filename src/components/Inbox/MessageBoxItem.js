import React, { Component } from 'react';
import { withStyles } from '@material-ui/styles';
import {
  Grid,
  Card,
  Typography,
  Avatar,
  Checkbox
} from "@material-ui/core"

const styles = theme => ({
  root: {
    padding: theme.spacing(2),
    marginBottom: theme.spacing(2)
  },
  avatarWrap: {
    padding: 8
  }
})

export class MessageBoxItem extends Component {
  render() {
    const { classes } = this.props
    return (
      <Card className={classes.root}>
        <Grid container justify="center" spacing={1}>
          <Grid item xs={2} style={{ padding: 8 }}>
            <Avatar src={require("../../assets/images/alice.png")} />
          </Grid>
          <Grid item xs={7}>
            <Typography variant="h6">Dr. Anita Ramos</Typography>
            <label variant="subtitle1">Re. John Smith</label>
          </Grid>
          <Grid item xs={2}>
            <label>11:30am</label>
          </Grid>
          <Grid item xs={1}>
            <Checkbox color="primary" style={{ padding: 0 }} />
          </Grid>
        </Grid>
        <Typography color="textSecondary" style={{ marginTop: 10 }}>
          Looks like the patient is suffering from macular edema caused by their diabetes. Order OCT to confirm.
        </Typography>
      </Card>
    )
  }
}

export default withStyles(styles)(MessageBoxItem)