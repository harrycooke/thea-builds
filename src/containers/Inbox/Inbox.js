import React, { Fragment } from 'react';
import { Redirect, Link } from 'react-router-dom';
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import MySnackbarContentWrapper from '../../components/SharedComponents/SnackBar';
import {
  Card,
  Grid,
  Typography,
  Button,
  Hidden,
  MenuItem,
  TextField,
  Divider,
  InputBase,
  InputAdornment,
  Snackbar,
} from '@material-ui/core';
import Media from 'react-media';
import SearchIcon from '@material-ui/icons/Search';
import AddIcon from "@material-ui/icons/Add";
import CreateIcon from '@material-ui/icons/Create';

import _ from "../../@lodash";
import { fetchAllCases, setCurrentCase, modifyCase, setSnackbar } from "../../actions/caseAction";
import { fetchAllDrafts, setCurrentDraft, removeDraft } from "../../actions/draftActions";
import MessageList from "../../components/Inbox/MessageList";
import styles from "../../assets/styles/InboxStyles";
import { specialtyList } from "../../utils/constant";
import ConfirmationDialog from "../../components/SharedComponents/ConfirmationDialog"
import { removeFileFromS3 } from '../../utils/s3';
import CircularProgress from '../../components/SharedComponents/CircularProgress';

import { sendAmplitudeData } from '../../utils/amplitude';

const AdapterLink = React.forwardRef((props, ref) => <Link innerRef={ref} {...props} />);

class Inbox extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      redirectToChat: false,
      redirectToAssignConsult: false,
      cases_id: null,
      specialty: null,
      redirectToCreateCase: false,
      readAllCases: [],
      readAllDrafts: [],
      selectedSpecialty: '',
      searchResultsForCases: [],
      searchResultsForDrafts: [],
      confirmDraftDelete: false,
      files: []
    }
    this.removeDraft = this.removeDraft.bind(this)
    this.openDraftDialog = this.openDraftDialog.bind(this)
    this.handleClose = this.handleClose.bind(this)
  }

  handleTrClicked = (id, isDraft, caseData) => {
    if (isDraft) {
      this.setState({
        redirectToCreateCase: true,
        selectedClientID: id
      });
      this.props.setCurrentDraft({ case_id: id })
    } else {
      let newCaseData = {
        caseId: id,
        last_updated_by_id: this.props.userdata.id
      }
      if(this.props.userdata.id === caseData.created_by_id){
        newCaseData['is_read'] = true;
      }else{
        let readByOthers = caseData.is_read_by_other ? caseData.is_read_by_other : [];
        let ifIndex = _.indexOf(readByOthers, this.props.userdata.id);
        if(ifIndex < 0){
          readByOthers.push(this.props.userdata.id);
          newCaseData['is_read_by_other'] = readByOthers;
        }
      }
      this.props.modifyCase(newCaseData);
      this.setState({
        redirectToChat: true,
        selectedClientID: id
      });
    }
  }

  handleSearch = e => {
    e.preventDefault();
    this.setState({
      [e.target.name]: e.target.value
    })
    let selectedSpecialty = e.target.value;
    const { readAllCases, readAllDrafts } = this.state;
    if (selectedSpecialty !== '') {
      let searchResultsForCases = readAllCases;
      let searchResultsForDrafts = readAllDrafts;
      // debugger;
      if(selectedSpecialty != "All Specialties"){                                         //Need to bug-proof here. If the text value to if condition is changed, the condition will break and no value returned.
        searchResultsForCases = _.filter(readAllCases, function (item) {
          return item.caseData.specialty.toLowerCase().indexOf(selectedSpecialty.toLowerCase()) > -1 || item.caseData.patient_name.toLowerCase().indexOf(selectedSpecialty.toLowerCase()) > -1;
        });
        searchResultsForDrafts = _.filter(readAllDrafts, function (item) {
          return item.draftData.specialty.toLowerCase().indexOf(selectedSpecialty.toLowerCase()) > -1 || item.draftData.patient_name.toLowerCase().indexOf(selectedSpecialty.toLowerCase()) > -1;
        });  
      }
      this.setState({ searchResultsForCases, searchResultsForDrafts });
    } else {
    // else if (selectedSpecialty == '') {
    //   this.setState({
    //     searchResultsForCases: readAllCases,
    //     searchResultsForDrafts: readAllDrafts
    //   });
    // }
    //   else {
      this.setState({   
        searchResultsForCases: readAllCases,
        searchResultsForDrafts: readAllDrafts
      });
    }
  }

  componentDidMount() {
    const currentUserData = {
      currentUserId: this.props.userdata.id,
      currentUserRole: this.props.userdata.user_type,
      isDeleted: false,
      isCompleted: false
    }
    sendAmplitudeData('Open Inbox page', currentUserData)
    setTimeout(() => {
      this.props.fetchAllCases(currentUserData);
      this.props.userdata.user_type === 'PCP' && this.props.fetchAllDrafts(currentUserData);
    }, 500)
  }

  UNSAFE_componentWillReceiveProps(nextProps) {
    const { readAllCases, readAllDrafts, errors } = nextProps;
    if (errors) {
      this.setState({
        errors: errors
      });
    }
    this.setState({
      readAllCases,
      readAllDrafts,
      searchResultsForCases: readAllCases,
      searchResultsForDrafts: readAllDrafts
    })
    if ((nextProps.isDraftDeleted !== this.props.isDraftDeleted) && nextProps.isDraftDeleted) {
      const currentUserData = {
        currentUserId: this.props.userdata.id,
        currentUserRole: this.props.userdata.user_type,
        isDeleted: false,
        isCompleted: false
      }
      setTimeout(() => {
        this.props.fetchAllCases(currentUserData);
        this.props.userdata.user_type === 'PCP' && this.props.fetchAllDrafts(currentUserData);
      }, 500)
    }
  }
  removeDraft() {
    this.setState({ confirmDraftDelete: false })
    const { filesDraft, draftId } = this.state;
    if (filesDraft && filesDraft.length) {
      filesDraft.map(async (fileObj, index) => {
        if (fileObj && fileObj.name) {
          return await removeFileFromS3(fileObj, index);
        }
      })
    }
    this.props.removeDraft({ case_id: draftId })

  }
  handleClose() { this.setState({ confirmDraftDelete: false }) }

  handleSnackbarClose = () => { 
    this.props.setSnackbar(false);
  }

  openDraftDialog(draftData) {
    this.setState({ confirmDraftDelete: true, draftId: draftData.case_id, filesDraft: draftData.files && draftData.files.length ? draftData.files : [] })
  }

  handleAssignConsult = (id, specialty) => {
    if(id){
      this.setState({redirectToAssignConsult: true, cases_id: id, specialty: specialty});
    }
  }

  render() {
    const { classes, userdata, readAllDrafts } = this.props;
    const {
      redirectToChat,
      redirectToCreateCase,
      selectedClientID,
      selectedSpecialty,
      searchResultsForCases,
      searchResultsForDrafts,
      redirectToAssignConsult,
      cases_id,
      specialty
    } = this.state;
    if (redirectToChat) {
      return (<Redirect to={{
        pathname: `/insidemessage/room_${selectedClientID}`,
        state: { from: 'cases' }
      }} />)
    }
    if (redirectToCreateCase) {
      return (<Redirect to={{
        pathname: `/edit-new-case-${selectedClientID}`,
      }} />)
    }
    if(redirectToAssignConsult){
      return (<Redirect to={{
        pathname: `/assign-consult/case_${cases_id}/specialty_${specialty}`,
      }} />)
    }

    let rows = [...searchResultsForDrafts, ...searchResultsForCases];

    return (
      <div className={classes.container}>
        <CircularProgress isLoading={this.props.draftLoader} />
        <CircularProgress isLoading={this.props.caseLoader} />
        {/* 
        <div>
          <Media queries={{
            small: "(max-width: 599px)",
            medium: "(min-width: 600px) and (max-width: 1199px)",
            large: "(min-width: 1200px)"
          }}>
            {matches => (
              <Fragment>
                {matches.small && <p>I am small!</p> && <style>  </style>}
                {matches.medium && <p>I am medium!</p>}
                {matches.large && <p>I am large!</p>}
              </Fragment>
            )}
          </Media>
        </div> */}

        {/* <Hidden lgUp>
          <Grid container spacing={4} style={{ paddingBottom: 20 }}>
            <Grid item xs={5}
              style={{
                flexDirection: "row",
                display: "flex",
                justifyContent: "flex-start"
              }}
            >
              <Button
                color="primary"
                component={AdapterLink}
                to="/inbox/cases"
                variant="contained"
                style={{
                  fontSize: 16,
                  width: 130,
                  // margin: 5
                }}>
                Open Cases
              </Button>
              <Button
                color="default"
                component={AdapterLink}
                to="/inbox/closed_cases"
                variant="contained"
                style={{
                  fontSize: 16,
                  backgroundColor: "#ffffff",
                  width: 130,
                  marginLeft: 20,
                }}>
                Closed Cases
              </Button>
              {readAllDrafts && readAllDrafts.length ?
                <Button
                  color="default"
                  component={AdapterLink}
                  to="/inbox/drafts"
                  variant="contained"
                  style={{
                    fontSize: 16,
                    backgroundColor: "#ffffff",
                    width: 80,
                    marginLeft: 32,
                  }}>
                  Drafts
              </Button> : null}
            </Grid>
            <Grid item xs={7}
              style={{
                display: "flex",
                justifyContent: "flex-end"
              }}
            >
              {userdata.user_type === "PCP" && (
                <Button
                  color="primary"
                  component={AdapterLink}
                  to="/create-new-case"
                  variant="contained"
                  style={{
                    fontSize: 16,
                    marginLeft: 32,
                    height: 45
                  }}>
                  Create a new Case
              </Button>)}
            </Grid>
          </Grid>
        </Hidden> */}
        <Grid container justify="flex-start" spacing={0} className={classes.cardContainer} >
          <Grid item xs={12}>
            <Card style={{
              backgroundColor: "#ffffff",
              padding: 20,
              overflow: "auto",
              height: "100%"
            }}>
              <Grid  container justify="space-between" >
                <Grid style={{ paddingTop: 6, paddingLeft: 15, paddingRight: 15, paddingBottom: 20,
                  display: "flex",
                  flexDirection: "row",
                  justifyContent: "flex-start"}}>
                  <Typography
                    color="textSecondary"
                    variant="h5"
                    fontWeight="100"
                  >
                    Open Cases
                  </Typography>
                  <Divider />
                </Grid>
                <Grid item xs={12}
                  sm={12} 
                  md={4} 
                  lg={4} 
                  xl={4} 
                  style={{
                  display: "flex",
                  flexDirection: "row",
                  justifyContent: "flex-start"
                }}>
                  <TextField
                    fullWidth
                    select
                    id="outlined-adornment-password"
                    className={classes.searchBar}
                    variant="outlined"
                    type={'text'}
                    size="small"
                    label="Filter by Specialty"
                    //label="Specialty type"
                    name={"selectedSpecialty"}
                    value={selectedSpecialty}
                    onChange={this.handleSearch}
                    style={{
                      display: userdata.user_type === "Admin" || userdata.user_type === "PCP"
                        ? "flex"
                        : "none"
                    }}
                  >
                    {specialtyList.map(specialty => (
                      <MenuItem key={specialty.value} value={specialty.value}>{specialty.label}</MenuItem>
                    ))}
                  </TextField>
                </Grid>
                <Grid item xs={12}
                  sm={12} 
                  md={4} 
                  lg={4} 
                  xl={4}   
                  style={{
                  display: "flex",
                  flexDirection: "row",
                  justifyContent: "flex-start"
                }}> 
                    <TextField
                      fullWidth
                      variant="outlined"
                      placeholder="Search"
                      size="small"
                      className={classes.searchBar}
                      InputProps={{
                       startAdornment: (
                         <InputAdornment position="start">
                           <SearchIcon  color="action"/>
                         </InputAdornment>
                       ),
                      }}
                      onChange={this.handleSearch}
                    />
                </Grid>
              </Grid>
              <MessageList
                openDraftDialog={this.openDraftDialog}
                rows={rows}
                isTrClicked={this.handleTrClicked}
                readAllCases={searchResultsForCases}
                readAllDrafts={searchResultsForDrafts}
                assignConsult={this.handleAssignConsult}
              />
            </Card>
            
          </Grid>
        </Grid>
        <ConfirmationDialog
          message="Are you sure you want to delete this draft?"
          title="Delete Draft"
          okOperationDialog={this.removeDraft}
          handleClose={this.handleClose}
          dialogState={this.state.confirmDraftDelete} />

          <Snackbar
            anchorOrigin={{
              vertical: 'bottom',
              horizontal: 'left',
            }}
            open={this.props.showSnackbar}
            autoHideDuration={3000}
            onClose={this.handleSnackbarClose}
          >
            <MySnackbarContentWrapper
              onClose={this.handleSnackbarClose}
              variant="success"
              message="Case sent successfully."
            />
          </Snackbar>
      </div>
    );
  }
}

Inbox.propTypes = {
  userdata: PropTypes.object.isRequired,
  fetchAllCases: PropTypes.func.isRequired,
  setCurrentCase: PropTypes.func.isRequired,
  modifyCase: PropTypes.func.isRequired,
  fetchAllDrafts: PropTypes.func.isRequired,
  setCurrentDraft: PropTypes.func.isRequired,
  readAllCases: PropTypes.array.isRequired
}

const mapStateToProps = state => ({
  userdata: state.auth.userdata,
  readAllCases: state.case.readAllCases,
  readAllDrafts: state.draft.readAllDrafts,
  isDraftDeleted: state.draft.isDraftDeleted,
  draftLoader: state.draft.draftLoader,
  caseLoader: state.case.caseLoader,
  showSnackbar: state.case.showSnackbar
})

export default connect(
  mapStateToProps,
  { fetchAllCases, setCurrentCase, modifyCase, fetchAllDrafts, setCurrentDraft, removeDraft, setSnackbar }
)(withStyles(styles)(Inbox))