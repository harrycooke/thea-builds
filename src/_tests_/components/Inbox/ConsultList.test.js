import React from 'react';
import ReactDOM from'react-dom';
import TestRenderer from 'react-test-renderer'; // ES6

import {ConsultList} from '../../../components/Inbox/ConsultList';
import sinon from 'sinon';
import { mount } from 'enzyme';

describe('ConsultList', () => {
    // beforeAll(() => {
    //     this.commonProps = {};
    // });
    
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(ConsultList, div);
    });

    it('renders React components, traversing a ReactElement tree and represents as JSON object', () => {
      const renderer = TestRenderer.create(ConsultList);
      renderer.toJSON();
    });

    describe('ConsultList', () => {
      it('should render correctly in "debug" mode', () => {
        const component = (ConsultList);
        expect(component).toMatchSnapshot();
      });
    });


    // it('calls funcName', () => {
    //     sinon.spy(ConsultList.prototype, 'funcName');
    //     const wrapper = mount(<ConsultList />);
    //     expect(ConsultList.prototype.componentDidMount).to.have.property('callCount', 1);
    //   });
    
    //   it('allows us to set props', () => {
    //     const wrapper = mount(<ConsultList bar="baz" />);
    //     expect(wrapper.props().bar).to.equal('baz');
    //     wrapper.setProps({ bar: 'foo' });
    //     expect(wrapper.props().bar).to.equal('foo');
    //   });
    
    //   it('simulates click events', () => {
    //     const onButtonClick = sinon.spy();
    //     const wrapper = mount((
    //       <ConsultList onButtonClick={onButtonClick} />
    //     ));
    //     wrapper.find('button').simulate('click');
    //     expect(onButtonClick).to.have.property('callCount', 1);
    //   });
});


