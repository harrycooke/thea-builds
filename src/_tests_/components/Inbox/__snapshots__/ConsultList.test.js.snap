// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConsultList ConsultList should render correctly in "debug" mode 1`] = `[Function]`;
